File: weekly-contest/weekly-contest-173/2-filter-restaurants-by-vegan-friendly-price-and-distance/sort.cpp
class Solution {
public:
    vector<int> filterRestaurants(vector<vector<int>>& restaurants, int veganFriendly, int maxPrice, int maxDistance) {
        const int n = restaurants.size();

        // 筛选
        vector<vector<int>> arr;
        for (int i = 0; i < n; ++i) {
            auto curr = restaurants[i];
            if (veganFriendly && !curr[2]) continue;
            if (curr[3] > maxPrice || curr[4] > maxDistance) continue;
            arr.push_back(curr);
        }

        // 排序
        sort(arr.begin(), arr.end(), [&](const auto &A, const auto &B) {
            if (A[1] != B[1]) return A[1] > B[1]; // rating高的
            return A[0] > B[0]; // id大的
        });

        // 取id返回
        vector<int> res;
        for (auto &r : arr) {
            res.push_back(r[0]);
        }
        return res;
    }
};

File: biweekly-contest/biweekly-contest-51/3/sort.js
/**
 * 排序
 */
var maximumElementAfterDecrementingAndRearranging = function(arr) {
  arr.sort((a, b) => a - b);
  const n = arr.length;
  if (arr[0] > 1) {
    arr[0] = 1;
  }
  for (let i = 1; i < n; ++i) {
    if (arr[i] - arr[i - 1] > 1) {
      arr[i] = arr[i - 1] + 1;
    }
  }
  return arr[n - 1];
};